### PR TITLE
Averages classes

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Make classes for averages (`XAverage`, `YZAverage`, ...) so that they can be recognized more easily that trying to find a division of integrals

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/expression_tree/averages.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+        self.f_a_dist = f_a_dist
+
+    def _unary_new_copy(self, child):
+        """See :meth:`UnaryOperator._unary_new_copy()`."""
+        return size_average(child, f_a_dist=self.f_a_dist)
+
+
+def x_average(symbol):
+    """
+    convenience function for creating an average in the x-direction.
```
Comment (brosaplanella) [inline pybamm/expression_tree/averages.py:95]:
```suggestion
    Convenience function for creating an average in the x-direction.
```

Not sure if what I am suggesting goes against the standard format. If not, note that the change also applies to all the other functions below.

File: pybamm/expression_tree/averages.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+            )
+        ):
+            aux = {"secondary": symbol.auxiliary_domains["secondary"]}
+            if "quaternary" in symbol.auxiliary_domains:
+                aux["tertiary"] = symbol.auxiliary_domains["quaternary"]
+            return pybamm.FullBroadcast(symbol.orphans[0], symbol.broadcast_domain, aux)
+        else:  # pragma: no cover
+            # It should be impossible to get here
+            raise NotImplementedError
+    # If symbol is a concatenation of Broadcasts, its average value is its child
```
Comment (brosaplanella) [inline pybamm/expression_tree/averages.py:158]: Probably a very dumb question but I don't get why that is the case should you take the average of each child and then average them together? (maybe it is what you are doing already, but didn't understand it).

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks Tino! Just a couple of minor comments. You also need to add a line to CHANGELOG before merging.

File: pybamm/expression_tree/averages.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/expression_tree/averages.py:158]: That is what I'm doing, I'll change the comment

**Pull request merged**